cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,441 @@
+
+### User scenarios
+
+#### User Story 1
+
+As a user I should be able to configure Radius AWS provider with IRSA through interactive experience.
+
+```
+% rad init --full  
+Enter an environment name 
+>default 
+
+% rad init --full  
+Select your cloud provider                     
+1. Azure                                  
+> 2. AWS                                     
+3.[back] 
+
+% rad init --full  
+
+Select the identity for the AWS cloud provider 
+1. IAM access key                              
+> 2. IAM Role for Service Account (IRSA)
+
+% rad init --full  
+
+Enter the IAM RoleARN for configuring IRSA: <role-arn-radius>
+```
+
+#### User story 2
+
+As a user I should be able to configure Radius AWS provider with IRSA using non interactive commands.
+
+1. Install the `rad` CLI.
+
+2. Install Radius using `rad install kubernetes`
+    ```
+    rad install kubernetes --set global.awsIRSA.enabled=true #default false
+    ```
+    If enabled, it should print a success/ failure message after updating the pod spec to mount the token. It should also print a information message to the complete of configuration by using rad credential register and rad env update.
+
+3. Create a Radius environment and add an AWS cloud provider.
+    ```
+    rad group create default
+
+    rad env create default
+    
+    rad env update default --aws-account-id <aws-account-id> --aws-region <aws-region> 
+    ```
+
+4. Register the role ARN using `rad credential register aws irsa`.
+    ```
+   rad credential register aws irsa --iam-role <roleARN>
+    ```
+
+Radius is now configured to use AWS IRSA for authentication.
+
+## Design
+
+### High Level Design for IRSA
+
+Users must choose whether or not to enable IRSA during Radius installation. Then the user should be able to enter the roleARN associated with desired policies which will be stored as the AWS credential.
+
+At a high level, the full flow to setup IRSA looks as below:
+
+1. configure the cluster with oidc provider. 
+2. have the role-arn configured with desired policies ready.
+3. take steps to allow  `radius-role-arn` and `eks cluster : namespace: service` to trust each other using aws portal.
+4. use [interactive](#user-story-1) approach or [non-interative](#user-story-2) to configure Radius to use IRSA.
+5. during rad deploy, radius assumes the role that is configured in the credential. 
+
+OIDC is configured for a cluster and not specific to a role-arn. Therefore, once we have support for multiple crdentials in Radius, the same solution would work for supporting multitenancy.
+
+More details about each step is covered in [Detailed Design](#detailed-design)
+
+IRSA works the same way on EKS as well as non-EKS clusters.  
+
+### Architecture Diagram
```

Comment:
Is there something we put together for this to show the call flow between components and to AWS?